api/services/SiteCreator.js

```diff
@@ -73,7 +73,7 @@ const createSiteFromExistingRepo = ({ siteParams, user }) => {
   const { owner, repository } = siteParams
 
   return Site.findOne({
-    where: { owner, repository },
+    where: { owner: owner.toLowerCase(), repository: repository.toLowerCase() },
```

Comment:
I'm a little weirded out we understand the owner and repo are always lowercased, but this file doesn't also contain the code enforcing said behavior. This is a split responsibility which would be easier to maintain if it were unified.

Is that an easy possibility?